src/idpi/grib_decoder.py

```diff
@@ -92,38 +128,49 @@ def load_data(
             if _is_ensemble(field)
             else ("step", "level")
         )
-        field_map[field.metadata(*dim_keys)] = field.to_numpy(dtype=np.float32)
+        key = field.metadata(*dim_keys)
+        field_map[key] = field.to_numpy(dtype=np.float32)
+
+        step = key[-2]  # assume all members share the same time steps
+        if step not in time_meta.get(param, {}):
+            time_meta.setdefault(param, {})[step] = field.metadata(namespace="time")
 
         if param not in dims:
             dims[param] = dim_keys[:-1] + (field.metadata("typeOfLevel"), "y", "x")
 
         if param not in metadata:
-            metadata[param] = field.metadata(
-                namespace=["ls", "geography", "parameter", "time"]
-            )
+            metadata[param] = field.metadata(namespace=["geography", "parameter"])
 
         if hcoords is None and param == ref_param:
             hcoords = {
                 dim: (("y", "x"), values) for dim, values in field.to_points().items()
             }
 
+        if extract_pv is not None and pv is None and param == extract_pv:
+            # assume pv is constant in time and ensemble perturbations
+            pv = field.metadata("pv")
+
     if not set(params) == data.keys():
         raise RuntimeError(f"Missing params: {set(params) - data.keys()}")
 
     result = {}
     for param, field_map in data.items():
         coords, shape = _gather_coords(field_map, dims[param])
+        tcoords = _gather_tcoords(time_meta[param])
         result[param] = xr.DataArray(
             np.array([field_map.pop(key) for key in sorted(field_map)]).reshape(shape),
-            coords=coords | hcoords,
+            coords=coords | hcoords | tcoords,
             dims=dims[param],
             attrs=metadata[param],
         )
-    return result
+    return result | _extract_pv(pv)
```

Comment:
update is in-place and returns None

Reply:
what is the downside of an in-place update for a return value?

Reply:
I generally try to avoid mutating variables in-place except when it provides a clear improvement in terms of performance or readability. Otherwise, I tend to think that mutating variables hinders readability.

Reply:
interesting. I think both approaches work fine here - mostly curiosity